src/models/flight_control/FGLinearActuator.cpp

```diff
@@ -0,0 +1,252 @@
+ * COMMENTS, REFERENCES,  and NOTES
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * 
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * INCLUDES
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+#include "FGLinearActuator.h"
+#include "input_output/FGXMLElement.h"
+#include <iostream>
+
+using namespace std;
+
+namespace JSBSim {
+    
+    /*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+     * CLASS IMPLEMENTATION
+     * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+    
+    FGLinear_Actuator::FGLinear_Actuator(FGFCS* fcs, Element* element) : FGFCSComponent(fcs, element)
+    {
+        initialized = 1;
+        versus = 0.0;
+        isSetProperty = false;
+        isResetProperty = false;
+        direction = 0;
+        countSpin = 0;
+        input_prec = 0.0;
+        module = 1.0;
+        rate = 0.3;
+        minRate = -0.3;
+        maxRate = 0.3;
+        lag = 0.0;
+        lagPrevius = 0.0;
+        gain = 1.0;
+        
+        if (element->FindElement("set")) {
+            setProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("set"));
+            isSetProperty = true;
+        }
+        if (element->FindElement("reset")) {
+            resetProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("reset"));
+            isResetProperty = true;
+        }
+
+        if (element->FindElement("versus")) versus = element->FindElementValueAsNumber("versus");
+        if (element->FindElement("lag")) lag = element->FindElementValueAsNumber("lag");
+        if (element->FindElement("module")) module = element->FindElementValueAsNumber("module");
+        if (element->FindElement("rate")) {
+            rate = element->FindElementValueAsNumber("rate");
+            if (rate <= 0 or rate > 1.0) rate = 0.5;
+            minRate = -(rate);
+            maxRate = rate;
+        }
+        if (element->FindElement("gain")) gain = element->FindElementValueAsNumber("gain");
+        
+        FGFCSComponent::bind();
+        
+        Debug(0);
+    }
+    
+    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    FGLinear_Actuator::~FGLinear_Actuator()
+    {
+        Debug(1);
+    }
+    
+    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    bool FGLinear_Actuator::Run(void )
+    {
+        Input = InputNodes[0]->getDoubleValue();
+        if (initialized) {
+            input_prec = Input;
+            initialized = 0;
+            std::cerr << "FGLinear_Actuator Cicle_zero " << InputNodes[0]->GetNameWithSign() << std::endl;
+            return true;
+        }
+        
+        Output = 0.0;
+        
+        set = 1.0;
+        reset = 0.0;
+        
+        if (isSetProperty) set = setProperty;
```

Comment:
You can save the use of the boolean `isSetProperty`. Just make sure that `setProperty` is initialized to `null` during construction.
```suggestion
        if (setProperty) set = setProperty;
```

Reply:
The question:
        if (isSetProperty) set = setProperty;
        if (setProperty) set = setProperty;
Currently setProperty (and resetProperty) are binary values, but they are likely to become analog values as they can be an easy way to trigger a fuzzy feature for this object. For this I used a Boolean variable to indicate to the system that the object operates in Set-Reset mode and eventually use the Set-Reset values as analog data to set the initial output value (resetProperty) or setProperty to store the value only when a certain threshold is exceeded. The isSetProperty (or isResetProperty) method makes these properties similar to a three-state logic in which the static configuration is provided with set and reset if these are connected (isSet ... true) or a logic without a set and reset if they are not connected (isSet or isReset false).

Reply:
Your statement is excellent and I very willingly follow the advice. So your correction is fine for me and I will be careful to avoid inserting the non-operative code in pull request, even if I plan to use it in the future.